data-serving/data-service/src/model/case.ts

```diff
@@ -1,43 +1,289 @@
 import mongoose from 'mongoose';
+import { ObjectId, Double } from 'mongodb';
 
-export const validOutcomes = ['pending', 'recovered', 'death'];
+export enum Sex {
+    Female,
+    Male,
+    Other,
+}
 
-const demographicsSchema = new mongoose.Schema({
-    age: {
+export enum Outcome {
+    Pending,
+    Recovered,
+    Death,
+}
+
+export enum Species {
+    HomoSapien,
+}
+
+export enum TravelPurpose {
+    Family,
+    Conference,
+    Work,
+}
+
+const dateValidator = {
+    type: Date,
+    min: '2019-11-01',
+    max: Date.now,
+};
+
+const rangeSchema = new mongoose.Schema({
+    minimum: {
+        type: Number,
+        min: -1,
+        max: 200,
+    },
+    maximum: {
         type: Number,
         min: 0,
         max: 200,
     },
+});
+
+const ageSchema = new mongoose.Schema({
+    years: {
+        type: Number,
+        min: -0.75,
+        max: 200,
+    },
```

Comment:
I assumed that these would be normalized ranges (e.g. 50-59), and we'd either only have those, when that's the best granularity available, or the actual years, from which we'd impute the range for ease of analysis.

But this might be an incorrect assumption... wdyt?

Reply:
Ah. Something to follow up with Stephen on.

If it's normalized ranges, I'm not sure this representation is right, though. Can we do something like a number enum set with the decade (0, 10, 20, 30, [...])?

Reply:
SG -- will follow up with Stephen on this!